exchangesync/node_policy.go

```diff
@@ -330,8 +332,10 @@ func UpdateNodePolicy(pDevice *persistence.ExchangeDevice, db *bolt.DB, nodePoli
 		return fmt.Errorf("Node policy does not validate. %v", err)
 	}
 
+	hasHwId := nodePolicy.Properties.HasProperty(externalpolicy.PROP_NODE_HARDWAREID)
```

Comment:
This is the code that I think is not checking if the node already has a hardware id before generating a new one.